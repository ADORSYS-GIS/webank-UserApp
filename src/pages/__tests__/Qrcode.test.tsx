import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import QRGenerator from "../Qrcode";
import { QRCodeCanvas } from "qrcode.react";
import { useLocation } from "react-router-dom";

// Mock QRCodeCanvas component
vi.mock("qrcode.react", () => ({
  QRCodeCanvas: vi.fn(() => <canvas data-testid="qrcode-canvas" />),
}));

// Mock useLocation hook from react-router-dom
vi.mock("react-router-dom", () => ({
  useLocation: vi.fn(),
}));

describe("QRGenerator Component", () => {
  const mockTotalAmount = "100";
  const mockAccountID = "12345ABC";
  const expectedQrValue = JSON.stringify({
    accountId: mockAccountID, // Update based on new component key
    amount: mockTotalAmount,
  });

  beforeEach(() => {
    vi.clearAllMocks();

    // Mock useLocation to return predefined state values
    (useLocation as jest.Mock).mockReturnValue({
      state: { totalAmount: mockTotalAmount, accountId: mockAccountID },
    });
  });

  it("renders QR code with correct values", () => {
    render(<QRGenerator />);

    // Check if QRCodeCanvas is called with correct props
    expect(QRCodeCanvas).toHaveBeenCalledWith(
      expect.objectContaining({
        value: expectedQrValue,
        size: 250,
      }),
      expect.anything(),
    );
  });

  it("handles back button click correctly", () => {
    const mockBack = vi.fn();
    vi.stubGlobal("history", { back: mockBack });

    render(<QRGenerator />);

    const backButton = screen.getByRole("button", { name: /back to entry/i });
    fireEvent.click(backButton);

    expect(mockBack).toHaveBeenCalledOnce();
  });
});
