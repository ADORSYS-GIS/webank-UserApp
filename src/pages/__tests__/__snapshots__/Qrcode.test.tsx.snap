// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QRGenerator Component > matches snapshot 1`] = `
<div>
  <div
    class="min-h-screen bg-gradient-to-r from-indigo-50 via-blue-100 to-indigo-50 flex items-center justify-center p-6"
  >
    <div
      class="bg-white rounded-2xl shadow-xl p-12 w-full max-w-md"
    >
      <h2
        class="text-4xl font-extrabold text-center text-gray-800 mb-10"
      >
        Top-Up QR Code
      </h2>
      <div
        class="flex flex-col items-center bg-white p-8 rounded-lg shadow-lg mb-8"
      >
        <canvas
          data-testid="qrcode-canvas"
        />
      </div>
      <div
        class="mt-6 flex justify-center"
      >
        <button
          class="px-6 py-3 text-lg font-medium text-white bg-blue-600 rounded-lg shadow-lg transform transition duration-300 hover:bg-blue-700 hover:scale-105 focus:outline-none"
        >
          ‚Üê Back to Entry
        </button>
      </div>
    </div>
  </div>
</div>
`;
